blocks/filter.py

```diff
@@ -52,10 +53,17 @@ class VariableFilter(object):
         ``False``, a variable matching any of the roles given will be
         returned. ``False`` by default.
     name : str, optional
+        The variable name. The Blocks name (i.e.
+        `x.tag.name`) is used.
+    name_re : str, optional
         A regular expression for the variable name. The Blocks name (i.e.
         `x.tag.name`) is used.
-    application : :class:`.Application` instance
+    applications : :class:`.Application` instance
+        or list of :class:`.Application` instance, optional
         Matches a variable that was produced by the application given.
+    one : boolean, optional
```

Comment:
Maybe something like `match_exactly_one`, `match_one` or `exactly_one` would make more sense. `one` is very cryptic if you don't know the docstring.